lib/widgets/message_list.dart

```diff
@@ -503,10 +513,52 @@ class MarkAsReadWidget extends StatelessWidget {
                   .merge(weightVariableTextStyle(context, wght: 400))),
                 shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(7)),
               ),
-              onPressed: () => _handlePress(context),
+              onPressed: () => loading ? null : _handlePress(context),
               icon: const Icon(Icons.playlist_add_check),
-              label: Text(zulipLocalizations.markAllAsReadLabel))))),
-    );
+              label: Text(zulipLocalizations.markAllAsReadLabel))))));
+  }
+}
+
+class MarkAsReadAnimation extends StatefulWidget {
+  const MarkAsReadAnimation({
+    super.key,
+    required this.loading,
+    required this.hidden,
+    required this.child
+  });
+
+  final bool loading;
+  final bool hidden;
+  final Widget child;
+
+  @override
+  State<MarkAsReadAnimation> createState() => _MarkAsReadAnimationState();
+}
+
+class _MarkAsReadAnimationState extends State<MarkAsReadAnimation> {
+  bool _isPressed = false;
+
+  void _setScale(bool isPressed) {
+    setState(() {
+      _isPressed = isPressed;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return GestureDetector(
+      onTapDown: (_) => _setScale(true),
+      onTapUp: (_) => _setScale(false),
+      onTapCancel: () => _setScale(false),
+      child: AnimatedScale(
+        scale: _isPressed ? 0.95 : 1,
+        duration: const Duration(milliseconds: 100),
+        curve: Curves.easeOut,
+        child: AnimatedOpacity(
+          opacity: widget.hidden ? 0 : widget.loading ? 0.55 : 1,
```

Comment:
This changes the appearance of the button depending on the loading state. Let's make that change in the `Handle loading state` commit, since otherwise the loading state isn't fully handled in that commit.

Also, is there a reason to choose 0.55 instead of 0.5, as Vlad mentioned in the [discussion](https://chat.zulip.org/#narrow/stream/48-mobile/topic/messages.20shift.20on.20mark-as-read/near/1813212)?

Reply:
Also, let's separate the on-pressed animation and the fade-to-hidden animation into separate commits, too.

In the commit for the on-pressed animation, let's remove the existing press animation, which is Material Design's "ink splash" effect; see #417. That can be done by adding `splashFactory: NoSplash.splashFactory` to the `FilledButton.styleFrom` call.